File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/crypto/impl/ApacheCodecBase64.java
package at.tugraz.tc.cyfile.crypto.impl;

import at.tugraz.tc.cyfile.crypto.Base64;

public class ApacheCodecBase64 implements Base64 {
    @Override
    public byte[] decode(String encoded) {
        return org.apache.commons.codec.binary.Base64.decodeBase64(encoded);
    }

    @Override
    public String encode(byte[] bytes) {
        return org.apache.commons.codec.binary.Base64.encodeBase64String(bytes);
    }
}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/crypto/impl/ApacheCodecBase64.kt
package at.tugraz.tc.cyfile.crypto.impl

import at.tugraz.tc.cyfile.crypto.Base64

class ApacheCodecBase64 : Base64 {
    override fun decode(encoded: String): ByteArray {
        return org.apache.commons.codec.binary.Base64.decodeBase64(encoded)
    }

    override fun encode(bytes: ByteArray): String {
        return org.apache.commons.codec.binary.Base64.encodeBase64String(bytes)
    }
}
